xapian-core/weight/bm25weight.cc

```diff
@@ -35,11 +35,23 @@ using namespace std;
 
 namespace Xapian {
 
+BM25Weight::BM25Weight(bool& bm25+)
+{
+	if(bm25+ != 0 && bm25+ != 1)
```

Comment:
Indeed, this check is pointless.

I'd avoid explicitly comparing boolean variables against an explicit value - instead of:

```
if (bm25_plus == true) {
    // Do something
}
if (flag == false) {
    // Do something else
}
```

write like so:

```
if (bm25_plus) {
    // Do something
}
if (!flag) {
    // Do something else
}
```

The intent seems clearer (at least to me) without the explicit test, and the behaviour is less surprising if the type of `bm25_plus` is `int` not `bool` - the first version checks in the value is `1`, the second if it is non-zero.

If you're dealing with C code using `int` for flags with something like:

```
#define FALSE 0
#define TRUE 1
```

then `if (value)` generally behaves more helpfully than `if (value == TRUE)` if `value` is set to (say) `2`.